src/itest/java/uk/gov/companieshouse/disqualifiedofficers/search/steps/DisqualificationSearchSteps.java

```diff
@@ -0,0 +1,115 @@
+
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static org.assertj.core.api.Assertions.assertThat;
+
+
+public class DisqualificationSearchSteps {
+
+    @Value("${disqualified-officers.search.topic.main}")
+    private String mainTopic;
+
+    @Value("${wiremock.server.port}")
+    private String port;
+
+    @Value("${disqualified-officers.search.group-id}")
+    private String groupId;
+
+    private static WireMockServer wireMockServer;
+
+    @Autowired
+    private KafkaTemplate<String, ResourceChangedData> kafkaTemplate;
+    @Autowired
+    private Logger logger;
+
+    private String type;
+    private String officerId;
+    private TestData testData = new TestData();
+
+    public static final String DISQUALIFICATION_RESOURCE_URI = "/disqualified-search/disqualified-officers/%s";
+
+
+    @Given("the application is running")
+    public void theApplicationRunning() {
+        assertThat(kafkaTemplate).isNotNull();
+    }
+
+    @When("the search consumer receives a {string} disqualification")
+    public void theConsumerReceivesDisqualificationOfType(String officerType) throws Exception {
+        configureWiremock();
+        this.officerId = "hv92jMgpl7e-ttvc5yZXqWiuHbQ";
+        this.type = officerType;
+        stubSearchApi();
+        ResourceChangedData data = testData.getResourceChangedData("src/itest/resources/input/" + officerType + "-disqualification.json");
+
+        kafkaTemplate.send(mainTopic, data);
+
+        CountDownLatch countDownLatch = new CountDownLatch(1);
+        countDownLatch.await(5, TimeUnit.SECONDS);
+        assertThat(wireMockServer.isRunning()).isTrue();
+    }
+
+    @Then("a PUT request is sent to the search api with the correct body")
+    public void putRequestIsSentToTheSearchApi() {
+        List<ServeEvent> serverEvents = getServeEvents();
+        assertThat(serverEvents.isEmpty()).isFalse();
+        assertThat(serverEvents.size()).isEqualTo(1);
+        assertThat(serverEvents.get(0).getRequest().getUrl()).isEqualTo(String.format(DISQUALIFICATION_RESOURCE_URI, this.officerId));
+        String expectedBody = "";
+        try {
+            expectedBody = testData.loadFile("src/itest/resources/output/" + type + "-output.json");
+        } catch(Exception e) {
+            logger.error("Failed to read file " + type + "-output.json: " + e.getMessage());
+        }
+        verify(1, putRequestedFor(urlMatching("/disqualified-search/disqualified-officers/" + officerId)));
+
+        List<ServeEvent> allServeEvents = getAllServeEvents();
+        ServeEvent serveEvent = allServeEvents.get(0);
+        String actualBody = serveEvent.getRequest().getBodyAsString();
+
+        assertThat(serveEvent.getResponse().getStatus()).isEqualTo(200);
+        assertThat(actualBody.replace(" ", ""))
```

Comment:
You could map both bodies to jackson trees and compare those for the json matching? That way you don't need to keep the property order exactly correct.

Reply:
Tried casting them both to objects but the match still failed whereas formatting the strings passed. Both objects should be identical but the assert didn't like them.

Reply:
Using readTree worked in the delta consumer e.g 
```
ObjectMapper mapper = new ObjectMapper();
JsonNode expectedTree= mapper.readTree(expectedBody);
JsonNode actualTree = mapper.readTree(actualBody);
assertEquals(expectedTree, actualTree)
```
I can see there are space differences in the output files e.g `"address_line_1":"ArnoldGardens",` maybe that's why the cast object didn't match?